R/utils-expr.R

```diff
@@ -963,3 +963,26 @@ polars_expr_to_r <- function(x) {
   }
   x
 }
+
+#' Timezone assertion
+#'
+#' @description `polars` tzone limitations
+#' https://pola-rs.github.io/r-polars/man/ExprDT_convert_time_zone.html
+#'
+#' @noRd
+#' @keywords internal
+check_timezone <- function(tz, null_allowed = FALSE) {
+  if (tz == "" & null_allowed) {
+    tz <- NULL
+    return(tz)
+  } else if (tz == "" & !null_allowed) {
```

Comment:
Use `&&` instead of `&`. Also, I'm not sure how `polars` behave when there are several timezones e.g.

```r
dat <- data.frame( x = ymd_hms(rep("2009-08-07 00:00:01", 2), tz = "America/New_York"))
dat |>
  mutate(x2 = force_tz(x, c("UTC", "Pacific/Auckland")))
```